src/master-playlist-controller.js

```diff
@@ -684,9 +684,18 @@ export class MasterPlaylistController extends videojs.EventTarget {
       }
     }
 
-    if (isEndOfStream) {
-      this.mediaSource.endOfStream();
+    const isUpdating = this.mediaSource.sourceBuffers
+      .filter((s) => s.updating === true).length > 0;
+    const isOpen = this.mediaSource.readyState === 'open';
+
+    // calling end of stream if the media source is not open
+    // or the sourcebuffers are updating causes an exception
+    // on chrome, this prevents that.
+    if (!isEndOfStream || isUpdating || !isOpen) {
+      return;
     }
+
+    this.mediaSource.endOfStream();
```

Comment:
Previously this function would cause an exception to be thrown on Chrome if the source buffers were updating or the media source was not open. This has been fixed.